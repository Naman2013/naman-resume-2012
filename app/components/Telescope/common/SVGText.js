import React from 'react';
import PropTypes from 'prop-types';
import { monoFont } from 'styles/variables/fonts';

const SVGText = ({
  text,
  x,
  y,
  displayProperties,
}) => (
  <g>
    <text
      style={Object.assign(displayProperties, SVGText.defaultProps.displayProperties)}
      x={x}
      y={y}
      alignmentBaseline="middle"
      textAnchor="middle"
    >
      {text}
    </text>
  </g>
);

SVGText.propTypes = {
  text: PropTypes.string.isRequired,
  x: PropTypes.number,
  y: PropTypes.number,
  displayProperties: PropTypes.shape({
    fill: PropTypes.string,
    fontFamily: PropTypes.string,
    fontSize: PropTypes.string,
  }),
};

SVGText.defaultProps = {
  x: 0,
  y: 0,
  displayProperties: {
    fill: 'aqua',
    fontFamily: monoFont,
    fontSize: '20px',
  },
};

export default SVGText;
