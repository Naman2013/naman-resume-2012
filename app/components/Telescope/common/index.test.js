import React from 'react';
import { shallow } from 'enzyme';
import SVGText from './SVGText';

describe('SVGText', () => {
  it('should render correctly', () => {
    const displayProperties = {
      fill: 'aqua',
      fontFamily: 'helvetica',
      fontSize: '20px',
    };
    const svgText = shallow(<SVGText text="This is a test..." x={100} y={100} displayProperties={displayProperties} />);
    expect(svgText).toMatchSnapshot();
  });
});
