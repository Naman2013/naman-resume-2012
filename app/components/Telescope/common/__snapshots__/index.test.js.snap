// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVGText should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SVGText
    displayProperties={
      Object {
        "fill": "aqua",
        "fontFamily": "monospace",
        "fontSize": "20px",
      }
    }
    text="This is a test..."
    x={100}
    y={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <text
        alignmentBaseline="middle"
        style={
          Object {
            "fill": "aqua",
            "fontFamily": "monospace",
            "fontSize": "20px",
          }
        }
        textAnchor="middle"
        x={100}
        y={100}
      >
        This is a test...
      </text>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alignmentBaseline": "middle",
        "children": "This is a test...",
        "style": Object {
          "fill": "aqua",
          "fontFamily": "monospace",
          "fontSize": "20px",
        },
        "textAnchor": "middle",
        "x": 100,
        "y": 100,
      },
      "ref": null,
      "rendered": "This is a test...",
      "type": "text",
    },
    "type": "g",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <text
          alignmentBaseline="middle"
          style={
            Object {
              "fill": "aqua",
              "fontFamily": "monospace",
              "fontSize": "20px",
            }
          }
          textAnchor="middle"
          x={100}
          y={100}
        >
          This is a test...
        </text>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alignmentBaseline": "middle",
          "children": "This is a test...",
          "style": Object {
            "fill": "aqua",
            "fontFamily": "monospace",
            "fontSize": "20px",
          },
          "textAnchor": "middle",
          "x": 100,
          "y": 100,
        },
        "ref": null,
        "rendered": "This is a test...",
        "type": "text",
      },
      "type": "g",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
